src/hostapi/jack/pa_jack.c

```diff
@@ -451,6 +452,43 @@ BlockingWaitEmpty( PaStream *s )
 
 /* ---- jack driver ---- */
 
+/* copy null terminated string source to destination, escaping regex characters with '\\' in the process */
+static void copy_string_and_escape_regex_chars( char *destination, const char *source, size_t destbuffersize )
+{
+    assert( destination != source );
+    assert( destbuffersize > 0 );
+
+    char *dest = destination;
+    /* dest_stop is the last location that we can null-terminate the string */
+    char *dest_stop = destination + (destbuffersize - 1);
+
+    const char *src = source;
+
+    while ( *src != '\0' && dest != dest_stop )
+    {
+        const char c = *src;
+        if ( strchr( "\\()[]{}*+?|$^.", c ) != NULL )
+        {
+            if( (dest + 1) == dest_stop )
+                break; /* only proceed if we can write both c and the escape */
+
+            *dest = '\\';
+            dest++;
+
+            *dest = c;
```

Comment:
This is redundant with the else branch

Reply:
Good point. I simplified this and force pushed.